cql3/restrictions/statement_restrictions.cc

```diff
@@ -563,22 +563,32 @@ std::pair<std::optional<secondary_index::index>, expr::expression> statement_res
     int chosen_index_score = 0;
     expr::expression chosen_index_restrictions;
 
-    for (const auto& index : sim.list_indexes()) {
-        auto cdef = _schema->get_column_definition(to_bytes(index.target_column()));
-        for (const expr::expression& restriction : index_restrictions()) {
-            if (has_partition_token(restriction, *_schema) || contains_multi_column_restriction(restriction)) {
-                continue;
-            }
-
-            expr::single_column_restrictions_map rmap = expr::get_single_column_restrictions_map(restriction);
-            const auto found = rmap.find(cdef);
-            if (found != rmap.end() && is_supported_by(found->second, index)
-                && score(index) > chosen_index_score) {
-                chosen_index = index;
-                chosen_index_score = score(index);
-                chosen_index_restrictions = restriction;
-            }
+    // Several indexes may be usable for this query. When their score is tied,
+    // let's pick one by order of the columns mentioned in the restriction
+    // expression. This specific order isn't important (and maybe in the
+    // future we could plan a better order based on the specificity of each
+    // index), but it is critical that two coordinators - or the same
+    // coordinator over time - must choose the same index for the same query.
+    // Otherwise, paging can break (see issue #7969).
+    for (const expr::expression& restriction : index_restrictions()) {
+        if (has_partition_token(restriction, *_schema) || contains_multi_column_restriction(restriction)) {
+            continue;
```

Comment:
An unrelated question, why are we calling 'continue' if restriction is a multi-column? What's wrong with using an index with multi-column restrictions?

Reply:
We don't handle indexes in cases like `(c1, c2, c3) = (1, 2, 3)`. Would be nice to have, but it's not that critical, the user can rewrite the query as `c1 = 1 AND c2 = 2 AND c3 = 3` and the index will be used.

I guess it's the easiest to just skip them. It's currently impossible to mix multi-column and single-column restrictions anyway.

Reply:
More importantly is to be able to use an index on c1,c2,c3 when only c1 is restricted as in c1=1

Reply:
There are no multi-column indexes at the moment